modules/drivers/bigquery/src/metabase/driver/bigquery/query_processor.clj

```diff
@@ -411,56 +426,42 @@
 ;;; |                                                Query Processor                                                 |
 ;;; +----------------------------------------------------------------------------------------------------------------+
 
-(defn- should-qualify-identifier?
-  "Should we qualify an Identifier with the dataset name?
-
-  Table & Field identifiers (usually) need to be qualified with the current dataset name; this needs to be part of the
-  table e.g.
-
-    `table`.`field` -> `dataset.table`.`field`"
-  [{:keys [identifier-type components] :as identifier}]
-  (cond
-    (::already-qualified? (meta identifier))
-    false
-
-    ;; If we're currently using a Table alias, don't qualify the alias with the dataset name
-    sql.qp/*table-alias*
-    false
-
-    ;; otherwise always qualify Table identifiers
-    (= identifier-type :table)
-    true
-
-    ;; Only qualify Field identifiers that are qualified by a Table. (e.g. don't qualify stuff inside `CREATE TABLE`
-    ;; DDL statements)
-    (and (= identifier-type :field)
-         (>= (count components) 2))
-    true))
-
 (defmethod sql.qp/cast-temporal-string [:bigquery :Coercion/YYYYMMDDHHMMSSString->Temporal]
   [_driver _coercion-strategy expr]
   (hsql/call :parse_datetime (hx/literal "%Y%m%d%H%M%S") expr))
 
-(defmethod sql.qp/->honeysql [:bigquery (class Field)]
-  [driver field]
-  (let [parent-method (get-method sql.qp/->honeysql [:sql (class Field)])
-        identifier    (parent-method driver field)]
-    (with-temporal-type identifier (temporal-type field))))
-
-(defmethod sql.qp/->honeysql [:bigquery Identifier]
-  [_ identifier]
-  (if-not (should-qualify-identifier? identifier)
+;; Table & Field identifiers (usually) need to be qualified with the current dataset name; this needs to be part of the
+;; table e.g.
+;;
+;;    "table"."field" -> "dataset.table"."field"
+(defn- qualify-identifier [identifier]
```

Comment:
It kinda seems qualifying identifiers is empirically part of almost every driver: maybe it would be better off as part of explicit driver interface. To be clear, this is an idle refactoring idea

Reply:
Yeah, there are enough drivers where we have to do something special here it probably would make sense.